src/Build/BuildCheck/Infrastructure/BuildCheckManagerProvider.cs

```diff
@@ -310,6 +325,14 @@ private void SetupAnalyzersForNewProject(string projectFullPath, BuildEventConte
                 }
             }
 
+            var infraStats = new Dictionary<string, TimeSpan>() {
+                { $"{BuildCheckConstants.infraStatPrefix}analyzerAcquisitionTime", _tracingReporter.analyzerAcquisitionTime },
+                { $"{BuildCheckConstants.infraStatPrefix}analyzerSetDataSourceTime", _tracingReporter.analyzerSetDataSourceTime },
+                { $"{BuildCheckConstants.infraStatPrefix}newProjectAnalyzersTime", _tracingReporter.newProjectAnalyzersTime }
+            };
+
+            _tracingReporter.TracingStats.Merge(infraStats, (span1, span2) => span1 + span2);
```

Comment:
This looks like should be responsibility of the `_tracingReporter` (as the individual times as well as the TracingStats property are all in its ownership).

That way we can in the future guard in case of concurrent access